RHDEVDOCS-3161 [RN] ImageChanges can now trigger builds based on information recorded in the BuildConfig status (#35169)
openshift/openshift-docs: release_notes/ocp-4-9-release-notes.adoc

[id="ocp-4-9-builds"]
=== Builds

[id="ocp-4-9-builds-image-changes"]
==== Use ImageChange in the BuildConfig status to trigger builds

> As a developer configuring GitOps deployments, you can add `ImageChange` triggers to the `BuildConfig` status. This type of trigger initiates a build when the base image it references is updated.

Reviewer: End users don't touch `status` in k8s objects - these should only be updated by controllers.

This feature means that developers can add BuildConfigs with ImageChange triggers to their OpenShift GitOps workflows. Image change triggers will not cause OpenShift GitOps to report itself as out of sync.

|`lastTriggeredImageID` field in the `BuildConfig` spec for Builds
|GA
|DEP

> |REM

Reviewer: We aren't really removing it, we are just ignoring the field and not updating it.

Author: @adambkaplan, to avoid introducing `IGN` (Ignored) as a new stage in the deprecation/removal lifecycle, I've reworded the description. WDYT?

Reviewer: Sounds good to me

[id="ocp-4-9-removed-features"]
=== Removed features

[id="ocp-4-9-builds-lasttriggeredimageid-parameter"]
==== The lastTriggeredImageID parameter in builds is removed

> This release removes the `lastTriggeredImageID` in the `ImageChangeTrigger` object, which is one of the `BuildTriggerPolicy` types that can be set on a `BuildConfig` spec.

Reviewer: We aren't removing the field, just ignoring it and not updating it.

```suggestion
This release stops updates to the `lastTriggeredImageID` in the `ImageChangeTrigger` object, which is one of the `BuildTriggerPolicy` types that can be set on a `BuildConfig` spec.
```

> Image change triggers do not start a build based on changes to the `lastTriggeredImageID` field in the `BuildConfig` spec. Instead, the image IDs that trigger a build are recorded in the status of the `BuildConfig` object.

Author: ```suggestion
derp derp derpta afkljasklfd asldfksd
```

> Image change triggers do not start a build based on changes to the `lastTriggeredImageID` field in the `BuildConfig` spec. Instead, the image IDs that trigger a build are recorded in the status of the `BuildConfig` object.

Author: ```suggestion
asdflkjads asldfkjl alfdkasjdlfj 
```

> Image change triggers do not start a build based on changes to the `lastTriggeredImageID` field in the `BuildConfig` spec. Instead, the image IDs that trigger a build are recorded in the status of the `BuildConfig` object, which most users cannot change.

Reviewer: Even cluster admins shouldn't change status fields. That is an easy way to break things.
```suggestion
Image change triggers do not start a build based on changes to the `lastTriggeredImageID` field in the `BuildConfig` spec. Instead, the image IDs that trigger a build are recorded in the status of the `BuildConfig` object.
```

Therefore, update scripts and jobs that inspect `buildConfig.spec.triggers[i].imageChange.lastTriggeredImageID` accordingly. (link:https://issues.redhat.com/browse/BUILD-190[*BUILD-190*])

> 

Author: @adambkaplan  How about using the following release note instead?
```
==== Cease updating the `lastTriggeredImageID` field and ignore it

The current release stops updating the `buildConfig.spec.triggers[i].imageChange.lastTriggeredImageID` field. Instead, this release updates the `buildConfig.status.imageChangeTriggers[i].lastTriggeredImageID` field.

Additionally, the `buildConfig.spec.triggers[i].BuildTriggerPolicy` ignores the `buildConfig.spec.triggers[i].imageChange.lastTriggeredImageID` field.

Now, you can configure the `buildConfig.spec.triggers[i].BuildTriggerPolicy` to start a build based on the `buildConfig.status.imageChangeTriggers[i].lastTriggeredImageID` field.

Therefore, update scripts and jobs that inspect `buildConfig.spec.triggers[i].imageChange.lastTriggeredImageID` accordingly. (link:https://issues.redhat.com/browse/BUILD-190[*BUILD-190*])
```
